src/components/CharacterTab.jsx

```diff
@@ -222,6 +232,33 @@ export default function CharacterTab() {
     cellStyle: { display: 'flex' },
   }), [])
 
+  const externalFilterChanged = useCallback((newValue) => {
+    setCurrentFilters(newValue)
+    gridRef.current.api.onFilterChanged()
+  }, [])
+
+  const charinfo = JSON.parse(JSON.stringify(DB.getMetadata().characters))
+
+  const doesExternalFilterPass = useCallback(
+    (node) => {
+      if (currentFilters.element.length && !currentFilters.element.includes(charinfo[node.data.id].element)) {
+        return false
+      }
+      if (currentFilters.path.length && !currentFilters.path.includes(ClassToPath[charinfo[node.data.id].path])) {
+        return false
+      }
+      if (!((charinfo[node.data.id].name).toLowerCase().includes(currentFilters.name) || (charinfo[node.data.id].displayName).toLowerCase().includes(currentFilters.name))) {
+        return false
+      }
+      return true
+    }, [charinfo, currentFilters],
+  )
+
+  const isExternalFilterPresent = useCallback(() => {
+    if (!currentFilters.element.length && !currentFilters.path.length && !currentFilters.name.length) return false
+    return true
+  }, [currentFilters])
```

Comment:
There's a very high performance impact of using [currentFilters] as a dependency here because this rerenders the entire character tab on every search bar keystroke and I'm getting some noticeable lag when typing in the bar. Probably needs an overall refactor to get this modularized enough so it doesn't do that. I can give it a shot unless you'd prefer to.